go/pkg/client/capabilities.go

```diff
@@ -28,6 +28,23 @@ func (c *Client) CheckCapabilities(ctx context.Context) (err error) {
 	if c.serverCaps.CacheCapabilities != nil {
 		c.MaxBatchSize = MaxBatchSize(c.serverCaps.CacheCapabilities.MaxBatchTotalSizeBytes)
 	}
+
+	if useCompression := c.CompressedBytestreamThreshold >= 0; useCompression {
```

Comment:
nit: `compressionSupported` as variable name?

Reply:
Technically, this is whether the _user_ requested to use Compression. 

The next few checks are the ones checking whether compression is supported :)

Reply:
Ah ok, then there isn't a need to use a variable here to explicitly identify it (it was a user specified option afterall). But tbh, I'm just ambivalent about it, so lgtm!

Reply:
it's purely for readability (particularly if someone is dropped in midst of the code base) - see bozydar's comment above

Reply:
Yep I saw that comment (which was what I disagreed with). I think the condition `c.CompressedBytestreamThreshold >= 0` is in itself very straightforward in stating that this block will be executed only if we are going to be using compression (there's very little room for misinterpretation).
Using a variable also seemed like a misdirection to me (I was searching for where this variable was being used again) - but again, this is all very subjective, which is why I didn't want to emphasize on it :)